### ISSUE TITLE
Clarify whether `Unsqueeze` should accept scalar `axes`

### ISSUE THREAD

**Issue by robertknight** [onnx/onnx#6886]
The specifications for Squeeze and Unsqueeze say that the `axes` input should be a "list of integers", which I took to mean a 1D tensor. ONNX Runtime expects the `axes` input to be a 1D tensor for [Squeeze](https://github.com/microsoft/onnxruntime/blob/d58ff6b9db9ec21db1be51874b4a5ba4787a6071/onnxruntime/core/providers/cpu/tensor/squeeze.h#L78) and [Reduce*](https://github.com/microsoft/onnxruntime/blob/d58ff6b9db9ec21db1be51874b4a5ba4787a6071/onnxruntime/core/providers/cpu/reduction/reduction_ops.cc#L755) ops, but for `Unsqueeze` specifically a change was made in https://github.com/microsoft/onnxruntime/pull/5665 to support scalars as well. The current test cases for Unsqueeze in the onnx repo always pass a vector for the `axes` input, although the reference implementation will accept either since numpy functions often allow `axis` arguments to be either a scalar or tuple.

I'm not sure what the preferred change is here, but if `Unsqueeze` _should_ accept scalar `axes`, then the spec should make this clear.

**Comment by justinchuby** [comment]
I don't think it should. In ONNX `axes` is always 1d and `axis` is always scalar.

**Comment by justinchuby** [comment]
https://github.com/onnx/onnx/pull/6888

**Comment by robertknight** [comment]
Thanks. I forgot to mention, the context where this issue came up was exporting a Swin Transformer model from timm, using the new TorchDynamo exporter:

```py
import timm
import torch

model = timm.create_model("swin_tiny_patch4_window7_224.ms_in1k", pretrained=True)
model = model.eval()
data_config = timm.data.resolve_model_data_config(model)
transforms = timm.data.create_transform(**data_config, is_training=False)
img = torch.zeros([1, 3, 224, 224])

onnx_prog = torch.onnx.export(model, img, dynamo=True)
onnx_prog.optimize()
onnx_prog.save("swin.onnx")
```

The resulting model worked in ORT but not some other runtimes that I tested, due to an Unsqueeze node (`node_Unsqueeze_43`) with scalar `axes` input.

**Comment by justinchuby** [comment]
I see. That we should fix. Thanks!
